src/theory/quantifiers/term_util.cpp

```diff
@@ -557,7 +557,7 @@ Node TermUtil::isSingularArg(Node n, Kind ik, unsigned arg)
   }
   else
   {
-    if (n.getType().isReal() && n.getConst<Rational>().sgn() < 0)
+    if (n.getType().isArithmetic() && n.getConst<Rational>().sgn() < 0)
```

Comment:
This should eventually become `n.getType().isInteger()`, no?

Reply:
True, I think we can change it now.